jax/numpy/lax_numpy.py

```diff
@@ -524,6 +524,21 @@ def _float_divmod(x1, x2):
 
 @_wraps(onp.power)
 def power(x1, x2):
+  # Special case for small positive integer scalars: use binary exponentiation.
+  # Using lax.pow may be imprecise for floating-point values; the goal of this
+  # code path is to make sure we end up with a precise output for the common
+  # pattern ``x ** 2`` or similar.
+  if isinstance(x2, int) and x2 >= 0 and x2 <= 64:
```

Comment:
You _could_ safely use Python's ternary comparison here:
```suggestion
  if isinstance(x2, int) and 0 <= x2 <= 64:
```

Reply:
Done.